pkg/build/generator/generator.go

```diff
@@ -218,9 +218,11 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 	if err != nil {
 		return nil, err
 	}
-
+	if request.Binary == nil && bc.Spec.Source.Dockerfile == nil && bc.Spec.Source.Git == nil && len(bc.Spec.Source.Images) == 0 {
```

Comment:
what validation?  client-side we don't even retrieve the buildconfig that i saw, so we can't do it there.

and the buildrequest object validation also doesn't currently have a need to retrieve the buildconfig object in order to do validation, it seems burdensome to introduce that call in the validation flow.